week-1-ts-ecosystem/w01e03-object-oriented-structure/src/models/shopping-basket.model.ts

```diff
@@ -0,0 +1,43 @@
+import { NewProductArgs, ProductType } from '../types';
+import { Product } from './product.model';
+import { v4 as uuidv4 } from 'uuid';
+
+export class ShoppingBasket<Type extends ProductType> {
```

Comment:
Generalnie pomysł z Generykiem aby odfiltrować jakeigo typu dane stosujemy, jest jak najbardziej ok.
Jednak zbyt tutaj w kodzie @michal-damiecki masz tych generyków.

Wydaje mi się że wystarczy jeśli `ShoppingBasket` byłby generycznym typem, jednak w tedy potrzebujesz jakiejś hierarchii klas na modelu `Product` jako klasie bazowej.

Jeśli nie, to sam Twój `enum` jako `ProductType` - mógłby posłużyć do pilnowania jakiego typu dane przechowujemy i wrzucamy do koszyka. Wtedy ten typ generyczny mógłby się okazać totalnie niepotrzebny.

Daj mi znać czy to tutaj widzisz - czy dać Ci jakiś przykład.